src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -260,34 +143,33 @@ function handleControlledInputBlur(inst, node) {
 var ChangeEventPlugin = {
   eventTypes: eventTypes,
 
-  _isInputEventSupported: isInputEventSupported,
-
-  extractEvents: function(
-    topLevelType,
-    targetInst,
-    nativeEvent,
-    nativeEventTarget,
-  ) {
+  extractEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget) {
     var targetNode = targetInst
       ? ReactDOMComponentTree.getNodeFromInstance(targetInst)
       : window;
 
+    // On the selectionchange event, the target is the document which isn't
+    // helpful becasue we need the input, so we use the activeElement instead.
+    if (!isTextInputEventSupported && topLevelType === 'topSelectionChange') {
+      targetNode = getActiveElement();
+
+      if (targetNode) {
+        targetInst = ReactDOMComponentTree.getInstanceFromNode(targetNode);
+      }
+    }
+
     var getTargetInstFunc, handleEventFunc;
+
     if (shouldUseChangeEvent(targetNode)) {
       getTargetInstFunc = getTargetInstForChangeEvent;
-    } else if (isTextInputElement(targetNode)) {
-      if (isInputEventSupported) {
-        getTargetInstFunc = getTargetInstForInputOrChangeEvent;
-      } else {
-        getTargetInstFunc = getTargetInstForInputEventPolyfill;
-        handleEventFunc = handleEventsForInputEventPolyfill;
-      }
-    } else if (shouldUseClickEvent(targetNode)) {
-      getTargetInstFunc = getTargetInstForClickEvent;
+    } else if (isTextInputElement(targetNode) && !isTextInputEventSupported) {
+      getTargetInstFunc = getTargetInstForInputEventPolyfill;
+    } else {
+      getTargetInstFunc = getTargetInstForInputOrChangeEvent;
     }
 
     if (getTargetInstFunc) {
-      var inst = getTargetInstFunc(topLevelType, targetInst);
+      var inst = getTargetInstFunc(topLevelType, targetInst, targetNode);
       if (inst) {
         var event = createAndAccumulateChangeEvent(
```

Comment:
If we use [this same check](https://github.com/facebook/react/pull/10238/files#diff-71f2afa49cd224b3a54e682f393c9c86R153) to branch when calling `createAndAccumulateChangeEvent` it should fix input backspacing in IE9 (tested locally).

````js
var event = createAndAccumulateChangeEvent(
  inst,
  nativeEvent,
  (!isTextInputEventSupported && topLevelType === 'topSelectionChange')
     ? targetNode
      : nativeEventTarget,
);
```

We can dedupe that check with a constant, but yeah. I think that should be work?